api/src/main/java/bio/terra/tanagra/service/search/Filter.java

```diff
@@ -74,5 +76,66 @@ public static ArrayFunction create(List<Filter> operands, Operator operator) {
     }
   }
 
-  // TODO implement a relationship filter tht allows chaining to another entity.
+  /**
+   * A {@link Filter} that introduces a constraint that there must be a relationship to another
+   * entity.
+   *
+   * <p>This quantifiers introduces a new {@link Variable} and implicitly binds it to an entity in
```

Comment:
grammar nit: "This quantifiers" I think is a typo, not sure what was intended though